lisa/tools/fio.py

```diff
@@ -0,0 +1,195 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import re
+from decimal import Decimal
+from enum import Enum
+from typing import Any, Dict, List, cast
+
+from lisa.executable import Tool
+from lisa.notifier import DiskPerformanceMessage
+from lisa.operating_system import Debian, Posix, Redhat, Suse
+from lisa.util import LisaException
+
+from .git import Git
+
+
+class FIOResult:
+    qdepth: int = 0
+    mode: str = ""
+    iops: Decimal = Decimal(0)
+    latency: Decimal = Decimal(0)
+    iodepth: int = 0
+
+
+FIOMODES = Enum(
+    "FIOMODES",
+    [
+        "randread",
+        "randwrite",
+        "read",
+        "write",
+    ],
+)
+
+
+class Fio(Tool):
+    fio_repo = "https://github.com/axboe/fio/"
+    branch = "fio-3.29"
+    # iteration21: (groupid=0, jobs=64): err= 0: pid=6157: Fri Dec 24 08:55:21 2021
+    # read: IOPS=3533k, BW=13.5GiB/s (14.5GB/s)(1617GiB/120003msec)
+    # slat (nsec): min=1800, max=30492k, avg=11778.91, stdev=18599.91
+    # clat (nsec): min=500, max=93368k, avg=130105.42, stdev=57833.37
+    # lat (usec): min=18, max=93375, avg=142.51, stdev=59.57
+    # get IOPS and total delay
+    _result_pattern = re.compile(
+        r"([\w\W]*?)IOPS=(?P<iops>.+?)," r"([\w\W]*?).* lat.*avg=(?P<latency>.+?),",
+        re.M,
+    )
+
+    @property
+    def command(self) -> str:
+        return "fio"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def install(self) -> bool:
+        posix_os: Posix = cast(Posix, self.node.os)
+        posix_os.install_packages("fio")
+        if not self._check_exists():
+            self._install_from_src()
+        return self._check_exists()
+
+    def launch(
+        self,
+        name: str,
+        filename: str,
+        mode: str,
+        iodepth: int,
+        numjob: int,
+        time: int = 120,
```

Comment:
@squirrelsc I changed runtime back to 120 seconds, previous value we used was 300, we did some comparison using different runtime, found 120 is the shortest runtime and the perf reaches the same effect when runtime is 300.